src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -94,15 +94,58 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_handle_short_am(MPIDI_OFI_am_header_t * m
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_HANDLE_SHORT_AM);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_HANDLE_SHORT_AM);
 
-    /* note: setting is_local, is_async, req to 0, 0, NULL */
-    MPIDIG_global.target_msg_cbs[msg_hdr->handler_id] (msg_hdr->handler_id, am_hdr,
-                                                       p_data, msg_hdr->data_sz, 0, 0, NULL);
+    /* note: setting is_local, is_async, req to 0, 1, rreq */
+    MPIDIG_global.target_msg_cbs[msg_hdr->handler_id] (msg_hdr->handler_id, am_hdr, p_data,
+                                                       msg_hdr->data_sz, 0, 0, NULL);
 
   fn_exit:
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_OFI_HANDLE_SHORT_AM);
     return mpi_errno;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_handle_pipeline(MPIDI_OFI_am_header_t * msg_hdr,
+                                                       void *am_hdr, void *p_data)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int is_done = 0;
+    MPIR_Request *rreq = NULL;
+    MPIR_Request *cache_rreq = NULL;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_HANDLE_PIPELINE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_HANDLE_PIPELINE);
+
+    cache_rreq = MPIDIG_req_cache_lookup(MPIDI_OFI_global.req_map, msg_hdr->sreq_ptr,
+                                         (uint64_t) msg_hdr->fi_src_addr);
+    MPL_DBG_MSG_FMT(MPIDI_CH4_DBG_GENERAL, VERBOSE,
+                    (MPL_DBG_FDEST, "cached req %p handle=0x%x", cache_rreq,
+                     cache_rreq ? cache_rreq->handle : 0));
+
+    rreq = cache_rreq;
+
+    if (!rreq) {
```

Comment:
Add comment: /* this is the first segment of a pipeline */

Reply:
ok